src/raygun.rum.js

```diff
@@ -155,12 +155,10 @@ var raygunRumFactory = function(window, $, Raygun) {
     };
 
     this.endSession = function() {
-      sendItemImmediately({
-        sessionId: self.sessionId,
-        requestId: self.requestId,
-        timestamp: new Date().toISOString(),
-        type: 'session_end',
-      });
+      self.sessionId = randomKey(32);
+      saveToStorage(self.sessionId);
```

Comment:
Would be good to have this functionality set in it's own private function and use that for both instances where we need to create a new session. That way any updates to this are bundled together.

Reply:
Abstracted into its own function